apps/doji-backend/src/friend/friend.service.ts

```diff
@@ -46,6 +46,32 @@ export class FriendService {
     }
   }
 
+  async getAllNotFriends(userRef: IUserReference) {
+    let friends = await this.getAllFriends(userRef)
+    let friendsUsername = []
+    friends.map((user) => {
+      friendsUsername.push(user.username)
+    })
+    // console.log(friendsUsername)
```

Comment:
```suggestion
    const friends = await this.getAllFriends(userRef)
    const friendsUsername = friends.map((user) => user.username) 
```

`friends.map((user) => user.username)` จะได้ออกมาเป็น Array of username

เพิ่มเติม

map function ทำหน้าที่แปลงของข้างใน array ทีละตัว (one by one) เช่น 

```tsx
const increaseByOne(item) {
  return item + 1
}
const arr = [1, 2, 3, 4]
const newArr = arr.map(increaseByOne)
console.log(newArr) // [2, 3, 4, 5]
```

หรือถ้าเขียนแบบ arrow function คือ

```tsx
const arr = [1, 2, 3, 4]
const newArr = arr.map((item) => item + 1)
console.log(newArr) // [2, 3, 4, 5]
```